### ISSUE TITLE
consistent-destructuring: incorrect report if the value is reassigned

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/eslint-plugin-unicorn#1407]
```js
const { children } = parent
parent.children = combine(children)
for(const child of parent.children) { // incorrect report here
}
```
